src/plugins/intel_cpu/src/nodes/rnn.cpp

```diff
@@ -863,8 +863,20 @@ void RNN::prepareParams() {
     prim = result.first;
 
     if (!wasMemoryPrepared || wFormatWasChanged) {
-        auto itpd = descs[0].createPrimitiveDescriptorIterator(getEngine(), dnnl::primitive_attr());
-        prepareMemory(itpd);
+        auto pd = (*prim).get_primitive_desc();
+        auto query_weights_md = [&pd](int idx = 0) -> dnnl::memory::desc {
+            auto what = dnnl::convert_to_c(dnnl::query::weights_md);
+            const dnnl_memory_desc_t *cdesc = dnnl_primitive_desc_query_md(pd, what, idx);
+            if (!cdesc)
+                IE_THROW() << "query_weights_md failed for idx " << idx << ".";
+            return dnnl::memory::desc(*cdesc);
+        };
+        std::vector<DnnlMemoryDescPtr> intDescs {
+            DnnlExtensionUtils::makeDescriptor(query_weights_md(0)),
+            DnnlExtensionUtils::makeDescriptor(query_weights_md(1)),
+            DnnlExtensionUtils::makeDescriptor(query_weights_md(2))
```

Comment:
My main concern about these line is that now we decouple our internal blobs reinitialization with that we have done in the constructor filling up `internalBlobDesc` vector. That means that now we have to align the code in both place if ne need to change the internal blobs content somehow. What do you think, m.b. it is better to query the primitive descriptor from the primitive and call `prepareMemory` from its iterator? So that we delete implicit coupling between this code frame and the internal blobs content. It should not add much overhead.

Reply:
I don't like this idea actually, I checked the whole repo, rnn is the only one that uses `internalBlobDesc` vector right now, I rather delete this `Node::prepareMemory(dnnl::primitive_desc_iterator& itpd)` overload  in this case since it's not general enough.

Also I found  a strange template function `createPrimitiveDescriptor()` which used by no one and it's calling this overload w/o initializing `internalBlobDesc`.

I rather use `Node::prepareMemory(const std::vector<DnnlMemoryDescPtr>& intDescs)` overload since deconv also use that and it's more versatile and flexible.

Reply:
I removed initialization of `internalBlobDesc` at the beginning of RNN constructor to make it more clear